logger/awslogs/logger.go

```diff
@@ -84,21 +87,27 @@ func (la *LoggerArgs) RunLogDriver(ctx context.Context, config *logging.Config,
 		logger.WithStream(stream),
 	)
 	if err != nil {
-		return errors.Wrap(err, "unable to create awslogs driver")
+		debug.LoggerErr  = errors.Wrap(err, "unable to create awslogs driver")
+		return debug.LoggerErr
 	}
 
 	if la.globalArgs.Mode == logger.NonBlockingMode {
-		debug.SendEventsToJournal(logger.DaemonName, "Starting non-blocking mode driver", journal.PriInfo)
+		debug.SendEventsToJournal(logger.DaemonName, "Starting non-blocking mode driver", journal.PriInfo, 0)
 		l = logger.NewBufferedLogger(l, la.globalArgs.MaxBufferSize)
 	}
 
 	// Start awslogs driver
-	debug.SendEventsToJournal(logger.DaemonName, "Starting awslogs driver", journal.PriInfo)
+	debug.SendEventsToJournal(logger.DaemonName, "Starting awslogs driver", journal.PriInfo, 0)
 	err = l.Start(la.globalArgs.UID, la.globalArgs.GID, la.globalArgs.CleanupTime, ready)
 	if err != nil {
-		return errors.Wrap(err, "failed to start awslogs driver")
+		debug.LoggerErr  = errors.Wrap(err, "failed to run awslogs driver")
+		// Do not return error if log driver has issue sending logs to destination, because if error
+		// returned here, containerd will identify this error and kill shim process, which will kill
+		// the container process accordingly.
+		// Reference: https://github.com/containerd/containerd/blob/release/1.3/runtime/v2/logging/logging.go
+		return nil
```

Comment:
No, shim logger will exit because this whole function returned here. It's just not returning error to Containerd shim and thus the container process will not be killed.